discovery-infra/test_infra/utils/utils.py

```diff
@@ -834,6 +834,39 @@ def fetch_url(url, timeout=60, max_retries=5):
     except (RequestException, HTTPError) as err:
         raise Exception(f"Failed to GET: {url} ({err})")
 
+def are_agents_in_status(
+        agnt, nodes_count, status,
+) -> bool:
+    agents = []
+    agents.append(agnt)
+    agents_in_status = [agent for agent in agents for condition in  agent.status()["conditions"] if condition["type"] == status]
```

Comment:
No need to create a list just for that
```suggestion
        agents, nodes_count, status,
) -> bool:
    agents_in_status = [agent for agent in agents for condition in  agent.status()["conditions"] if condition["type"] == status]
```